### ISSUE TITLE
RFC: Require changelog entries on PRs with code changes

### ISSUE THREAD

**Issue by ashfurrow** [artsy/artsy-danger#16]
## Proposal: 

Require changes to code to be accompanied by a change to a changelog entry (if applicable). 

## Reasoning

Many projects under the Artsy org have changelogs, and we are currently replicating Danger configs to require changelog entries ([example](https://github.com/artsy/eigen/blob/0a0dd66a98c1944598e78a3dd813a09d97d36340/Dangerfile#L12)). We should consider standardizing this across the org using Peril.

## Exceptions: 

If there's no changelog in the repo, the rule should not apply.

Additionally, the rule could be smart about what constitutes a "code change." For example, a PR that only adds unit tests shouldn't require a changelog entry. I don't have precise logic for what this would look like, but it would be doable with a few "if" statements I'm sure. We can also update the rule to handle edge cases as they occur.

**Comment by jonallured** [comment]
This seems like a slam dunk to me and I think you're right that no CHANGELOG file and a few if statements could bring some sanity. 🏀 🏀 🏀

**Comment by kanaabe** [comment]
> If there's no changelog in the repo, the rule should not apply. 

👍

**Comment by ashfurrow** [comment]
So I did some work on this yesterday and ran into a problem implementing the RFC. We want to avoid warning about a missing changelog entry if there's no changelog present in the repo, however, there's no supported way to get a "list of files in the repo" through the GitHub API. So we have no way of enforcing this:

> If there's no changelog in the repo, the rule should not apply.

Anyone have ideas for a workaround?

**Comment by orta** [comment]
Yep, just make a list of assumed files and check for their contents, roughly

```js

schedule(rfc("xx", "CHANGELOG", async () => {
  
  const getFile = danger.github.utils.fileContents
  const readmes = ["CHANGELOG.md", "changelog.md", "CHANGELOG.yml"]
  for (var readme of readmes) {
    const content = await getFile(readme)
    if (content) {
      // File exists
    }
  }

}))
```

**Comment by orta** [comment]
Fixed by #21
